### PR TITLE
Fix Onlyoffice

### PR DESCRIPTION
**Pull request by szaimen**
fix https://github.com/nextcloud/vm/issues/1018

### CODE REVIEW

**Comment by szaimen** [comment]
@enoch85 Actually, I haven't tested this check yet, but in theory it should work ;)

File: apps/onlyoffice.sh [hunk 1]
```diff
@@ -48,6 +48,18 @@ sudo bash onlyoffice.sh"
     exit 1
 fi
 
+# check if apache2 evasive-mod is enabled and disable it because of compatibility issues
+if [ "$(Apache2ctl -M | grep evasive)" != "" ]
+then
+    msg_box "Please note: if you continue, the ddos protection for webservices will get disabled now, because of compatibility issues with onlyoffice."
+    if [[ "no" == $(ask_yes_or_no "So do you want to continue?")  ]]
+        then
```
Comment (enoch85) [inline apps/onlyoffice.sh]: Skip the "So" here, I noticed you add that on all questions. 

It's enough with just "Do you want to continue?" :)

File: apps/onlyoffice.sh [hunk 2]
```diff
@@ -48,6 +48,18 @@ sudo bash onlyoffice.sh"
...
+            exit 1
```
Comment (enoch85) [inline apps/onlyoffice.sh]: Just add `sleep 0.1` here, if you exit the whole script exits.

File: apps/onlyoffice.sh (hunk 1 above)
Comment (enoch85) [reply apps/onlyoffice.sh]: I will fix it later, let it be, there are more changes to be made.

**Comment by enoch85** [comment]
@szaimen Please test if this works.

File: apps/onlyoffice.sh [hunk 3]
```diff
@@ -48,6 +48,20 @@ sudo bash onlyoffice.sh"
...
+if [ "$(apache2ctl -M | grep evasive)" != "" ]
+then
+    msg_box "We noticed that 'mod_evasive' is installed which is the DDOS protection for webservices. It has comptibility issues with OnlyOffice and $
+    if [[ "no" == $(ask_yes_or_no "Do you want to disable DDOS protection?")  ]]
+        then
+            sleep 0.1
+        else
+            a2dismod evasive
+	    a2dismod mod-evasive
+            apt -y install libapache2-mod-evasive
```
Comment (szaimen) [inline apps/onlyoffice.sh]: Axtually to my experiemce disabling 'a2dismod evasive' is enough. But don't know if mod-evasive also doesn't work then...

File: apps/onlyoffice.sh (hunk 1 above)
Comment (szaimen) [reply apps/onlyoffice.sh]: you have already done this 👍

File: apps/onlyoffice.sh (hunk 2 above)
Comment (szaimen) [reply apps/onlyoffice.sh]: you have already done this ..👍

**Comment by szaimen** [comment]
> @szaimen Please test if this works.

Yes, the check works. (currently line 52)

I have just tested  "a2dismod evasive" and this already makes sure that onlyoffice works. So the other removals aren't really much needed to make onlyoffice work...

File: apps/onlyoffice.sh (hunk 3 above)
Comment (enoch85) [reply apps/onlyoffice.sh]: Yeah OK, if you say so. :) The module activated in the Extra Security script is called `mod-evasive`. That's why I added it.

File: apps/onlyoffice.sh [hunk 4]
```diff
@@ -58,7 +58,7 @@ then
     else
         a2dismod evasive
         a2dismod mod-evasive
-        apt -y install libapache2-mod-evasive
+        apt purge libapache2-mod-evasive -y
```
Comment (enoch85) [inline apps/onlyoffice.sh]: Haha, what a mistake! :D

**Comment by enoch85** [comment]
> aren't really much needed to make onlyoffice work...

OK, great! only do what's necessary then. :+1:

File: apps/onlyoffice.sh (hunk 1 above)
Comment (enoch85) [reply apps/onlyoffice.sh]: It "will" not be disabled, the user can "choose" to disable it. There's a big difference.

Please change wording.

**Comment by enoch85** [comment]
OK, let's merge this!

**Comment by enoch85** [comment]
@szaimen FYI https://github.com/nextcloud/vm/issues/961#issuecomment-544281236

@BetaLeaf Do you think this will do the trick? Or do we need to disable `mod_evasive` as well?

**Comment by szaimen** [comment]
@enoch85
As @BetaLeaf has written above that comment:
> Now I am not knowledgeable enough to figure out how to set the server configuration properly, so I just disabled mod_evasive entirely and that seems to solve the issue.
> 
> ```
> sudo a2dismod evasive
> sudo systemctl restart apache2
> ```

That was the reason why I came up with this.

**Comment by szaimen** [comment]
I think 
> sudo a2dismod evasive
and
> sudo a2dismod mod_evasive 
Both disable mod_evasive.

**Comment by enoch85** [comment]
Yeah, my bad. Please ignore.

**Pull request merged**